apps/src/sites/hourofcode.com/pages/public/events/index.js

```diff
@@ -257,8 +257,14 @@ function validateFields() {
     } else {
       $('#school-name-error').hide();
     }
+  }
 
-    if ($('#hoc-event-location').val() === '') {
+  if (
+    ['out_of_school', 'after_school'].includes($('#hoc-event-type').val()) ||
+    ($('#hoc-event-type').val() === 'in_school' &&
+      ($('#country').val() !== 'US' || schoolData.nces === SCHOOL_NOT_FOUND))
+  ) {
+    if ($('.mapboxgl-ctrl-geocoder--input').val() === '') {
```

Comment:
`$('#hoc-event-location').val()` was not returning the current entry in the location field, but its nested `$('.mapboxgl-ctrl-geocoder--input').val()` was correctly obtaining the value.